pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -200,6 +202,28 @@ func (fs *Decomposedfs) ListStorageSpaces(ctx context.Context, filter []*provide
 		return spaces, nil
 	}
 
+	client, err := pool.GetGatewayServiceClient(fs.o.GatewayAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	checkRes, err := client.CheckPermission(ctx, &permissionsv1beta1.CheckPermissionRequest{
```

Comment:
Can we move this check to `gateway.ListStorageSpaces`? Then you won't need to duplicate it in every driver

Reply:
I agree that this would be nicer but it's not really possible since we need the permission information in the decomposedfs in line https://github.com/cs3org/reva/pull/2341/files/cc9def031675c4301ca9256cbf6fbfc8555fd71e#diff-d308a0558595437a537e77e29346472e66ab3bc31c213fcd2dd9b5f64c02aa91R384.

Sure this could be passed down via an `Opaque` field but that wouldn't really be secure since everybody could just put that field into the Opaque map.